private-media-framework/draft-ietf-perc-private-media-framework.md

```diff
@@ -711,3 +698,362 @@ a co-author.
+
+The endpoint would then take this packet and pass it to the code responsible
+for applying the SRTP cryptographic transform for PERC.  In Double
+[@!I-D.ietf-perc-double], this is looked at by the endpoint as a single
+operation, but since two different keys are utilized and since some readers
+will be responsible for writing that software, the steps are presented
+separately.
+
+The first step in applying the Double transform is to encrypt the packet
+using the endpoint's E2E SRTP master key.  This is the "inner" transform.
+Once that cryptographic transform has been applied using the E2E key, the
+packet would look like the one shown in [@fig-e2e-protected-packet].
+
+{#fig-e2e-protected-packet align="center"}
+~~~
++------------------------------+
+|          RTP Header          |
++------------------------------+
+|                              |
+|      Ciphertext Payload      |
+|                              |
++------------------------------+
+|      Authentication Tag      |
++------------------------------+
+~~~
+Figure: Packet after Cryptographic Transform using E2E Key
+
+Note that the E2E key is not used directly.  Rather, SRTP ([@RFC3711])
+specifies in Section 4.3.1 how to derive the session encryption key
+(k_e) from the SRTP master key.  It is k_e that is used to actually
+encrypt the packet.  This is true both this step and the next step where
+the HBH key is used.  To keep the text simple here, the reader should
+understand that E2E keys or HBH keys must be passed through the key
+derivation function specified in RFC 3711 or elsewhere depending on the
+cipher.
+
+With certain cryptographic transforms like AES-GCM ([@RFC7714]), use of
+a separate authentication tag is discouraged (as it is redundant).
+However, it is shown here for completeness.  When using AES-GCM, the
+authentication data is appended to the end of the ciphertext.  It is
+important to mention since Double specifies the use of AES-GCM and
+so PERC implementers will not see a separate field in the payload for
+authentication.
+
+### Step 3: Apply the HBH Cryptographic Transform
+
+Once the "inner" transform has been applied, the OHB RTP header
+extension specified in [@!I-D.ietf-perc-double] would be added or
+modified as appropriate.  Once that is done, the "outer" transform is
+applied.  For the "outer" transform, the HBH key is used and the
+resulting packet would look like [@fig-hbh-protected-packet].
+
+{#fig-hbh-protected-packet align="center"}
+~~~
++------------------------------+
+|          RTP Header          |
++------------------------------+
+|                              |
+|      Ciphertext Payload      |
+|                              |
+|        Encrypting both       |
+|      Ciphertext Payload      |
+|              and             |
+|      Authentication Tag      |
+|     (from previous step)     |
+|                              |
++------------------------------+
+|      Authentication Tag      |
++------------------------------+
```

Comment:
I find this really confusing - I think of there being one auth tag with two halfs and one cipher text. I think the thing that might help in above discution of both innner and outer is to make it clear what is encryped for both inner and outer and what is included in authenticated data for both steps. I'm not sure the auth tag handling is correct - will need to look at that.